packages/app/src/systems/Assets/components/AssetLogo/AssetLogo.tsx

```diff
@@ -0,0 +1,46 @@
+import { Image, Avatar } from '@fuel-ui/react';
+import { useMemo } from 'react';
+
+import type { Asset } from '../../services';
+import { getAssetEth, getAssetFuel } from '../../utils';
+
+type EthAssetLogoProps = {
+  asset?: Asset;
+  size?: number;
+  alt?: string;
+};
+
+const DEFAULT_SIZE = 18;
+
+export const AssetLogo = ({
+  alt = 'AssetLogo',
+  asset,
+  size = DEFAULT_SIZE,
+}: EthAssetLogoProps) => {
+  const { image, address } = useMemo(() => {
+    if (asset?.icon) {
+      return { image: asset.icon };
+    }
+
+    if (!asset) return {};
+
+    const ethAsset = getAssetEth(asset);
+    const fuelAsset = getAssetFuel(asset);
+
+    return {
+      address: ethAsset?.address || fuelAsset?.assetId || fuelAsset?.contractId,
```

Comment:
I think a contract should not be considered the address of the asset as it will never be true on the Fuel side.

```suggestion
      address: ethAsset?.address || fuelAsset?.assetId,
```